internal/app/components/users/users.go

```diff
@@ -1,33 +1,91 @@
 package users
 
 import (
+	"context"
 	"github.com/gzcharleszhang/course-planner/internal/app/components/courses"
 	"github.com/gzcharleszhang/course-planner/internal/app/components/timelines"
+	"github.com/gzcharleszhang/course-planner/internal/app/db"
 	"github.com/rs/xid"
+	"go.mongodb.org/mongo-driver/bson"
+	"golang.org/x/crypto/bcrypt"
 )
 
 type FirstName string
 type LastName string
 type UserId string
+type PasswordHash string
+
+type UserData struct {
+	Id            UserId                   `bson:"_id"`
```

Comment:
out of curiosity, how does `bson` here differ from `json`

Reply:
`bson` is like `json` but it's used by mongo. They store more types of data than `json` i think